packages/frontend/src/hooks/validation/useValidation.ts

```diff
@@ -73,3 +102,39 @@ export const useValidationMutation = (
         ),
     });
 };
+
+export const useValidationUserAbility = (projectUuid: string) => {
+    const { data: user } = useUser(true);
+    const { data: project } = useProject(projectUuid);
+    const canUserSeeValidationErrorsNotifications =
+        !!user &&
+        !!project &&
+        user.ability?.can(
+            'manage',
+            subject('Validation', {
+                organizationUuid: project.organizationUuid,
+                projectUuid,
+            }),
+        );
+    return canUserSeeValidationErrorsNotifications;
+};
+
+export const useValidationNotificationChecker = (): [boolean, () => void] => {
```

Comment:
The local storage is only updated when `useValidation` is triggered. That hook is only used in the validation page. We need to get the latest validation date in this hook and update the local storage.

Reply:
Yes, understand your point, but the hook is also used in `ValidationErrorNotification.tsx` so I guess it would trigger the notification check on `NotificationsMenu/index.tsx`, no? 🤔

Reply:
`ValidationErrorNotification ` is only rendered if `useValidationNotificationChecker` returns true.

Reply:
my bad, was getting confused about what hook was being used to conditionally render the  `NotificationsMenu` 👍